src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[3];
+  const arr = date.split(key);
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  let yearIndex = fromFormat.indexOf('YY');
+
+  if (yearIndex === -1) {
+    yearIndex = fromFormat.indexOf('YYYY');
+  }
+
+  const day = arr[dayIndex];
+  const month = arr[monthIndex];
+  let year = arr[yearIndex];
+
+  if (year.length === 2) {
+    if (year < 30) {
+      year = '20' + year;
+    } else {
+      year = '19' + year;
+    }
+  }
+
+  if (toFormat.includes('YY')) {
+    year = year.slice(-2);
+  }
+
+  switch (toFormat[0]) {
+    case 'DD':
+      return +day + toFormat[3] + month + toFormat[3] + year;
+    case 'MM':
+      return month + toFormat[3] + +day + toFormat[3] + year;
+    default:
+      return year + toFormat[3] + month + toFormat[3] + +day;
+  }
```

Comment:
Consider creating an empty array and pushing parts of the date here in the loop, to get the right order. After the loop try to join your array with the separator